cpp/config/Make.xcframework.rules

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) ZeroC, Inc. All rights reserved.
+
+# Creates a xcframework from a static component and its headers
+# $1: The name of the framework (Ice, IceDiscovery, etc.)
+# $2: The platform
+# $3: Output path
+define create-xcframework-from-library
+
+lib/$1.xcframework: lib/lib$1.a
+	$(E) Creating lib/$1.xcframework
+	$(Q)$(RM) -r lib/$1.xcframework
+	$(Q)if [ -d include/$1 ]; then \
+		xcodebuild -create-xcframework -library lib/lib$1.a -headers include/$1 -output lib/$1.xcframework ; \
+		cp include/generated/$1/*.h lib/$1.xcframework/macos-arm64/Headers ; \
+		mkdir -p lib/$1.xcframework/macos-arm64/Headers/$1 ; \
+		find lib/$1.xcframework/macos-arm64/Headers/* -prune ! -name $1 -exec mv {} lib/$1.xcframework/macos-arm64/Headers/$1 \; ; \
+	else \
+		xcodebuild -create-xcframework -library lib/lib$1.a -output lib/$1.xcframework; \
+	fi
+
+srcs all:: lib/$1.xcframework
+
+clean::
+	$(E) Cleaning lib/$1.xcframework
+	$(Q)$(RM) -r lib/$1.xcframework
+
+endef
+
+ifeq ($(filter $(CONFIGS),static),)
+Ice_always_enable_configs := static
+IceDiscovery_always_enable_configs := static
+IceLocatorDiscovery_always_enable_configs := static
```

Comment:
This causes the macOS builds to always build static libraries. In a follow-up PR I plan to move this deeper into the build system for macOS.